pkg/kubelet/kuberuntime/helpers.go

```diff
@@ -255,6 +255,46 @@ func (m *kubeGenericRuntimeManager) getSeccompProfile(annotations map[string]str
 	return ""
 }
 
+func fieldSeccompProfile(scmp *v1.SeccompProfile, profileRootPath string) *runtimeapi.SecurityProfile {
+	if scmp == nil {
+		return &runtimeapi.SecurityProfile{
+			ProfileType: runtimeapi.SecurityProfile_Unconfined,
+		}
+	}
+	if scmp.Type == v1.SeccompProfileTypeRuntimeDefault {
+		return &runtimeapi.SecurityProfile{
+			ProfileType: runtimeapi.SecurityProfile_RuntimeDefault,
+		}
+	}
+	if scmp.Type == v1.SeccompProfileTypeLocalhost && scmp.LocalhostProfile != nil && len(*scmp.LocalhostProfile) > 0 {
```

Comment:
should we use unconfined if local host is set and nil profile in this new mode.. or make this an error.. or keep the type and empty ref.. letting the caller decide how to handle the situation.

Reply:
This is mirroring the existing behavior. Added a TODO for moving to runtime default as a follow up.

Reply:
> Added a TODO for moving to runtime default as a follow up.

Aren't there compatibility considerations for that?

Reply:
> Aren't there compatibility considerations for that?

Yes, existing workloads could potentially break. Hence, we need to come up with a plan for defaulting
to runtime default instead of unconfined. We could potentially add a feature gate. I think it is best
tackled separately (from this PR) with wider discussion in sig-node first.

Reply:
Agreed. If you add a TODO, make sure it points to an issue with sufficient context to ensure no one tries to resolve the TODO without addressing those issues.

Reply:
Sure, I will open an issue for it :+1: